src/compiler/gcc.rs

```diff
@@ -303,6 +309,55 @@ where
         };
     }
 
+    let xclang_it = ExpandIncludeFile::new(cwd, &xclangs);
+    for arg in ArgsIter::new(xclang_it, (&ARGS[..], &clang::ARGS[..])) {
```

Comment:
It feels a little unfortunate to conflate the set of options that `clang` takes with the set of options that are valid for `-Xclang` since they're not exactly the same. However, there's enough overlap (and we're definitely relying on that overlap with some of the things we use in our clang-cl builds) that it would probably be a pain to split them out.